test/unit/APITest.php

```diff
@@ -43,17 +43,23 @@ private function getMockAndRegister($originalClassName, $methods) {
       return $mockObject;
     }
 
-    protected function setUp() {
-      RequestHandlerFactory::set_instance(NULL);
-    }
-
-    protected function tearDown() {
-      RequestHandlerFactory::set_instance(NULL);
+    public function mockApiResponse($expected_api_url, $expected_data, $response) {
```

Comment:
Can you make it private because it is not used at other files?

Reply:
Right, I will.

Reply:
Done, thanks for review.